### PR TITLE
ci: added tests to check if schema was changed since last release

### PR DESCRIPTION
**Pull request by CptSchnitz**

### CODE REVIEW

File: scripts/checkFilesChanged.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+if (addedAndModifiedFiles.length === 0) {
+  console.log('No schema files have been added or modified since the latest release.');
+  process.exit(0);
+}
+const errorHandler = new ErrorHandler();
+errorHandler.addError(
+  ...addedAndModifiedFiles.map(([action, file]) => ({
+    file,
+    error: `Schema files should not be ${action === 'A' ? 'added' : 'modified'} outside of a release.`,
```
Comment (shimoncohen) [inline scripts/checkFilesChanged.ts:24]: What about deleted? You consider that a modification?

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -22,3 +22,26 @@ jobs:
 
       - name: Run the validations
         run: npm run validate
+
+  check_for_changed_files:
```
Comment (shimoncohen) [inline .github/workflows/pull_request.yml:26]: Can't a release be after a few PR's then we decide to "seal" the version?
I feel like this is too strict.

Comment (CptSchnitz) [reply .github/workflows/pull_request.yml:26]: it compares to the last release, so how is it strict?

File: scripts/checkFilesChanged.ts (hunk 1 above)
Comment (CptSchnitz) [reply scripts/checkFilesChanged.ts]: fixed

File: .github/workflows/pull_request.yml (hunk 2 above)
Comment (shimoncohen) [reply .github/workflows/pull_request.yml:26]: I mean that it seems to not allow changes in schemas unless it's part of a new release, but maybe we should allow this in order to let people make changes until there's an agreed upon version.

**Pull request merged**